tests/nvme_test.py

```diff
@@ -160,8 +160,10 @@ def get_ctrl_id(self):
                                 encoding='utf-8')
         err = proc.wait()
         self.assertEqual(err, 0, "ERROR : nvme list-ctrl failed")
-        line = proc.stdout.readline()
-        ctrl_id = line.split(":")[1].strip()
+        lines = proc.stdout.readlines()
+        self.assertTrue(len(lines) > 1, "ERROR : nvme list-ctrl could not find ctrl")
+        # Skipping fist stdout line "num of ctrls present: <num_ctrls>"
+        ctrl_id = lines[1].split("]:")[1].strip()
```

Comment:
I think we should start using the json output of the commands, this would make all the crazy parsing unnecessary.

Reply:
I agree, the code is much cleaner like that. :)
For now, I start using the json output on the lines that I am touching within the commits.
I will submit a chore patchset for some cleanups where I transition the rest to use json outputs.